### PR TITLE
Version updates to Alpine, Golang and s3fs.

### PR DESCRIPTION
**Pull request by tristanmorgan**
Also slips in `-trimpath` into the compile flags.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -22,15 +23,16 @@ RUN apk --no-cache add \
  && make install \
  && strip /usr/bin/s3fs
 
-FROM golang:1.19-alpine as builder
+FROM golang:1.21-alpine as builder
 RUN apk add git make binutils
-COPY / /work
-WORKDIR /work
+COPY / /go/src/github.com/majst01/csi-driver-s3
```
[the remark this answers is not in the document]
Comment (tristanmorgan) [reply Dockerfile]: it was more of a habitual change. I have reverted back to the original.

P.S. the mailcap package is for S3fs to have access to mime types.

File: Dockerfile [hunk 2]
```diff
@@ -39,5 +41,5 @@ RUN apk --no-cache add \
     util-linux
 COPY --from=s3fs-builder /usr/bin/s3fs /usr/bin/s3fs
 RUN /usr/bin/s3fs --version
-COPY --from=builder /work/bin/s3driver /s3driver
+COPY --from=builder /go/src/github.com/majst01/csi-driver-s3/bin/s3driver /s3driver
```
Comment (majst01) [inline Dockerfile:44]: still

Comment (tristanmorgan) [reply Dockerfile:44]: 🤦 I’ll fix and push soon.

Comment (tristanmorgan) [reply Dockerfile]: fixed that too.

**Review by majst01 — APPROVED** [review]

**Pull request merged**